### PR TITLE
feat: expose accounts and store API endpoints

### PR DESCRIPTION
**Pull request by rostyk-kanafotskyy**
## Type

* ### Feature
PCM APIs exposed through JS-SDK

## Description
Added ability to use PCM APIs by creating alternative auth flow

Code for testing
```
  const authenticate = () => {
      const body = {
        grant_type: 'password',
        username: '',
        password: '',
        client_id: '',
        client_secret: ''
      }

      return fetch('https://api.moltin.com/oauth/access_token', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded'
        },
        body: Object.keys(body)
                .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(body[k])}`)
                .join('&')
      }).then(res => res.json())
    }

    const Moltin = moltin.gateway({
      authenticate
    })

    Moltin.Authenticate().then(res => console.log(res))
    Moltin.Accounts.All().then(({ data }) => console.log(data))
    Moltin.Accounts.Keys().then(({ data }) => console.log('Keys',data))
    Moltin.Accounts.Stores().then(({ data }) => {
      Moltin.Accounts.StoresSwitch(data[0].id).then(() => {
        Moltin.Products.All().then(({ data }) => console.log(data))
      })
    })
```

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ class Config {
       language,
       host,
       storage,
-      custom_fetch
+      custom_fetch,
+      authenticate
```
Comment (samblacklock) [inline src/config.js:15]: Maybe we could name this `custom_authenticator` to match the existing naming and to show that it is optional?

File: src/endpoints/accounts.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return this.request.send(
+      `${this.endpoint}/stores/${id}`,
+      'GET',
+      undefined,
+      undefined,
+      this
+    )
+  }
+
+  StoresSwitch(storeId) {
```
Comment (samblacklock) [inline src/endpoints/accounts.js:31]: `SwitchStore` may be a clearer name?

**Review by samblacklock — COMMENTED** [review]
Great work @rostyk-kanafotskyy, just a couple of suggestions 👆

Also, do you think you could add a few quick unit tests for the new `/accounts` methods?

File: src/endpoints/accounts.js (hunk 2 above)
Comment (rostyk-kanafotskyy) [reply src/endpoints/accounts.js:31]: Yes, I needed that.
Thanks, Sam
Fixing

**Review by shaunmaharaj — APPROVED** [review]

**Review by samblacklock — APPROVED** [review]
👍

**Pull request merged**